src/wifi/model/periodic-dmg-wifi-scheduler.cc

```diff
@@ -0,0 +1,524 @@
+{
+  static TypeId tid = TypeId ("ns3::PeriodicDmgWifiScheduler")
+    .SetParent<DmgWifiScheduler> ()
+    .SetGroupName ("Wifi")
+    .AddConstructor<PeriodicDmgWifiScheduler> ();
+  return tid;
+}
+
+PeriodicDmgWifiScheduler::PeriodicDmgWifiScheduler ()
+{
+  NS_LOG_FUNCTION (this);
+}
+
+PeriodicDmgWifiScheduler::~PeriodicDmgWifiScheduler ()
+{
+  NS_LOG_FUNCTION (this);
+}
+
+void
+PeriodicDmgWifiScheduler::DoDispose ()
+{
+  NS_LOG_FUNCTION (this);
+  DmgWifiScheduler::DoDispose ();
+}
+
+void
+PeriodicDmgWifiScheduler::UpdateStartAndRemainingTime (void)
+{
+  // for the periodic scheduler, m_allocationStartTime is useless, since the addition
+  // of new SPs is consecutive 
+  NS_LOG_FUNCTION (this);
+  if (m_addtsAllocationList.empty ())
+    {
+      // no existing allocations
+      m_remainingDtiTime = m_dtiDuration.GetMicroSeconds ();
+      m_availableSlots.push_back (std::make_pair (0, m_dtiDuration.GetMicroSeconds ()));
+    }
+  else
+    {
+      // if there are existing allocations, update DTI time just for consistency
+      m_remainingDtiTime = 0;
+      for (const auto & slot: m_availableSlots)
+        {
+          m_remainingDtiTime += (slot.second - slot.first);
+        }
+    }
+}
+
+void
+PeriodicDmgWifiScheduler::AdjustExistingAllocations (AllocationFieldListI iter, uint32_t duration, bool isToAdd)
+{
+  NS_LOG_FUNCTION (this << duration << isToAdd);
+
+  // This method is called upon a DelTsRequest or after the cleanup of 
+  // non-pseudostatic allocations.
+  // In this version of the periodic scheduler, existing allocations are not shifted 
+  // to fill the created gaps but only the vector listing the available slots is updated.
+  // For this reason, the current input parameters are useless.
+
+  auto addtsListCopy = m_addtsAllocationList;
+
+  // sort the copy to simplify the process of going through the allocation list
+  sort (addtsListCopy.begin (),
+        addtsListCopy.end (),
+        [](const AllocationField& lhs, const AllocationField& rhs){
+      return lhs.GetAllocationStart () < rhs.GetAllocationStart ();
+    });
+
+  uint32_t startSlot = 0;
+  std::vector<std::pair<uint32_t, uint32_t> > newDTI; 
+
+  for (const auto & allocation: addtsListCopy)
+    {
+      // this loop goes through the list of allocations, spotting all the available
+      // slots and adding them to the list
+      if (startSlot < allocation.GetAllocationStart ())
+        {
+          newDTI.push_back (std::make_pair (startSlot, allocation.GetAllocationStart ()));
+        }
+
+      startSlot = allocation.GetAllocationStart () + allocation.GetAllocationBlockDuration () + m_guardTime;
+    }
+
+  // the following check ensure that if there is a trailing available slot in the DTI,
+  // it will be correctly considered and added to the list.
+  if (startSlot < m_dtiDuration.GetMicroSeconds ())
+    {
+      newDTI.push_back (std::make_pair (startSlot,  m_dtiDuration.GetMicroSeconds ()));
+    }
+
+  m_availableSlots = newDTI;
```

Comment:
That's actually a good point, basically as I did in `AdjustExistingAllocation`. Initially I did it to avoid problems in case of concurrent access to `m_availableSlots`, but since we the scheduler is a single entity, we shouldn't have the problem. 

If also @mattia-lecci agrees, I'll do it as you suggested to avoid unnecessary memory occupation :)